pennylane/compiler/qjit_api.py

```diff
@@ -0,0 +1,113 @@
+# Copyright 2023 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""QJIT compatible quantum and compilation operations API"""
+
+from .compiler import CompileError, AvailableCompilers, available
+
+
+def qjit(fn=None, *args, compiler="catalyst", **kwargs):  # pylint:disable=keyword-arg-before-vararg
+    """A decorator for just-in-time compilation of hybrid quantum programs in PennyLane.
+
+    This decorator enables both just-in-time and ahead-of-time compilation,
+    depending on the compiler package and whether function argument type hints
+    are provided.
+
+    .. note::
+
+        Currently, only the :doc:`Catalyst <catalyst:index>` hybrid quantum-classical
+        compiler is supported. The Catalyst compiler works with the JAX interface
+
+        For more details, please see the Catalyst documentation and :func:`catalyst:.qjit`
+        docstring.
+
+    .. note::
+
+        Catalyst supports compiling QNodes that use ``lightning.qubit``,
+        ``lightning.kokkos``, ``braket.local.qubit``, and ``braket.aws.qubit``
+        devices. It does not support ``default.qubit``.
+
+        Please see the :doc:`Catalyst documentation <catalyst:index>` for more details on
+        supported devices, operations, and measurements.
+
+    Args:
+        compiler (str): name of the compiler to use for just-in-time compilation
+        fn (Callable): the quantum or classical function to compile
+        autograph (bool): Experimental support for automatically converting Python control
+            flow statements to Catalyst-compatible control flow. Currently supports Python ``if``,
+            ``elif``, ``else``, and ``for`` statements. Note that this feature requires an
+            available TensorFlow installation. Please see the
+            :doc:`AutoGraph guide <catalyst:dev/autograph>` for more information.
+        target (str): the compilation target
+        keep_intermediate (bool): Whether or not to store the intermediate files throughout the
```

Comment:
Where are the files stored?

Reply:
The files are stored at the location where the Python script is called.

Reply:
Fixed #4812